components/webext-storage/src/db.rs

```diff
@@ -105,6 +97,53 @@ impl DerefMut for StorageDb {
     }
 }
 
+// We almost exclusively use this SharedStorageDb
+pub struct SharedStorageDb {
```

Comment:
thought: `SharedStorageDb` doesn't feel like quite the right name ("What makes it shared? With whom? 'Shared' as opposed to what?" are a few questions that went through my mind reading this the first time), but I'm struggling to think of a more descriptive one! 😅

Maybe `InterruptibleStorageDb`? It looks like we have a similar concept of an `InterruptibleConnection` in the hand-written Kotlin bindings that's a thread-safe database connection + interrupt handle.

WDYT?

Reply:
The "shared" is really just to reflect the `Mutex<>` - what I don't particularly like about `InterruptibleConnection` is that the inner `StorageDb` protected by the mutex is *also* interruptible. Maybe `ConcurrentConnection`? Or the even more explicit `ThreadSafeConnection`?

Reply:
> Or the even more explicit `ThreadSafeConnection`?

Oh, I like that! `ThreadSafeStorageDb`?